### DISCUSSION TITLE
Smooth a voxelated boundary

### DISCUSSION THREAD

**Discussion by SunilAnandatheertha** [marcomusy/vedo#600] [Q&A]
Hi, I have a voxelated data as below.

![image](https://user-images.githubusercontent.com/66170566/154859587-8514219b-d426-4aa5-91f1-beb092a3fb9e.png)

I need to smooth the boundaries. I am also inclined to extract a surface for each of the bounding surface, like as is being done in the image in this [repo](https://github.com/SunilAnandatheertha/XtalMesh). Though I am coming up with my own stuff to extract and characterize the boundary of each voxel-set (i.e. iso-color voxels), I am wondering if this can also be done from vedo.

**Comment by marcomusy** [comment]
Hi, what kind of data is it? polygonal or volumetric?
you can use `threshold()` and `smooth()` methods... a small example dataset would help..
